library/ssl_tls.c

```diff
@@ -8761,6 +8812,66 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
         ssl->handshake = mbedtls_calloc( 1, sizeof(mbedtls_ssl_handshake_params) );
     }
 
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    /* If the buffers are too small - reallocate */
```

Comment:
@AndrzejKurek , I try to understand the flow of too small buffers. Could you explain when we will want to increase the buffers again ?

Reply:
It's before each handshake that we want to have full sized buffers. After the handshake they can be shrinked down again. Renegotiation also forces buffer resizing, as the MFL (maximum fragment length) might change, and this affects buffer sizes.